radixdlt-engine/src/main/java/com/radixdlt/application/tokens/construction/StakeTokensConstructorV3.java

```diff
@@ -65,21 +64,11 @@ public void construct(StakeTokens action, TxBuilder builder) throws TxBuilderExc
 			builder.up(new TokensInAccount(action.from(), REAddr.ofNativeToken(), change));
 		}
 
-		var flag = builder.read(
-			AllowDelegationFlag.class,
-			p -> p.getValidatorKey().equals(action.to()),
-			Optional.of(action.to()),
-			"Could not find state"
-		);
+		var flag = builder.read(AllowDelegationFlag.class, action.to());
 
 		if (!flag.allowsDelegation()) {
 			final REAddr owner;
-			var validator = builder.read(
-				ValidatorOwnerCopy.class,
-				p -> p.getValidatorKey().equals(action.to()),
-				Optional.of(action.to()),
-				"Could not find state"
-			);
+			var validator = builder.read(ValidatorOwnerCopy.class, action.to());
 			owner = validator.getOwner();
```

Comment:
The `owner` is not used until this point. Perhaps it is worth to declare it here and eliminate default initialization.

Reply:
good point